src/function-set.ts

```diff
@@ -24,9 +24,11 @@ export class FunctionSet {
 
 	constructor(
 		public readonly pattern: string,
+		public readonly distribution: string,
 		private readonly log: (message: string) => void,
 		public readonly origin: Origin = new Origin()
 	) {
+		this.distribution = distribution
```

Comment:
This line is unnecessary because TypeScript will automatically assign any `public` constructor argument as an instance value.
```suggestion
```